src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -1243,6 +1244,33 @@ namespace Partition {
     }
 } // namespace Partition
 
+namespace Type {
+
+    static inline ASR::expr_t *eval_Type(Allocator &al, const Location &loc,
+            Vec<ASR::expr_t*> &args) {
+        LCOMPILERS_ASSERT(ASRUtils::all_args_evaluated(args));
+        ASR::ttype_t* t = ASRUtils::expr_type(args[0]);
+        return ASR::down_cast<ASR::expr_t>(ASR::make_TypeConstant_t(al, loc, t));
+    }
+
+    static inline ASR::asr_t* create_Type(Allocator& al, const Location& loc,
+            Vec<ASR::expr_t*>& args,
+            const std::function<void (const std::string &, const Location &)> err) {
+
+        if (args.size() != 1) {
+            err("Intrinsic type function currently supports only 1 argument", loc);
+        }
+
+        // TODO - Replace this type , it shouldn't be integer .
+        ASR::ttype_t *type = ASRUtils::TYPE(ASR::make_Integer_t(al, loc,
+                            4, nullptr, 0));;
+
```

Comment:
Firstly here . Guessing what our IntrinsicFunction node should look like in the ASR 
```
                                        [(IntrinsicFunction
                                            Type
                                            [(StringConstant
                                                "hi"
                                                (Character 1 2 () [])
                                            )]
                                            0
                                            // Return Type of expected output
                                            //  Compile time value 
                                        )]
```
1) Here the expected ouput would be a `type` variable so we need return type of `type` and I'm not sure if we have a `ttype` for that so I've kept it as int as of now just to pass some checks .

2) Next we would get some compile time value from `eval_Type` and we would like to have an expr node to store this type , so I ended up introducing a `TypeConstant(ttype type)` in ASR.asdl to hold a constant type value .